EmailNotification/handler.ts

```diff
@@ -50,140 +49,141 @@ export type EmailNotificationResult = t.TypeOf<typeof EmailNotificationResult>;
   notificationDefaultParams: INotificationDefaults
-) => async (
-  context: Context,
-  input: unknown
-): Promise<EmailNotificationResult> => {
-  const inputOrError = EmailNotificationInput.decode(input);
-
-  if (E.isLeft(inputOrError)) {
-    context.log.error(
-      `${context.executionContext.functionName}|Cannot decode input`
-    );
-    context.log.verbose(
-      `${context.executionContext.functionName}|ERROR_DETAILS=${readableReport(
-        inputOrError.left
-      )}`
-    );
-    return EmailNotificationResult.encode({
-      kind: "FAILURE",
-      reason: "DECODE_ERROR"
-    });
-  }
-
-  const { notificationEvent } = inputOrError.right;
-
-  const {
-    message,
-    content,
-    notificationId,
-    senderMetadata
-  } = notificationEvent;
-
-  const logPrefix = `${context.executionContext.functionName}|MESSAGE_ID=${message.id}|NOTIFICATION_ID=${notificationId}`;
-
-  // Check whether the message is expired
-  const errorOrActiveMessage = ActiveMessage.decode(message);
-
-  if (E.isLeft(errorOrActiveMessage)) {
-    // if the message is expired no more processing is necessary
-    context.log.warn(`${logPrefix}|RESULT=TTL_EXPIRED`);
-    return EmailNotificationResult.encode({
-      kind: "SUCCESS",
-      result: "EXPIRED"
-    });
-  }
-
-  // fetch the notification
-  const errorOrMaybeNotification = await lNotificationModel.find([
-    notificationId,
-    message.id
-  ])();
-
-  if (E.isLeft(errorOrMaybeNotification)) {
-    const error = errorOrMaybeNotification.left;
-    // we got an error while fetching the notification
-    context.log.warn(`${logPrefix}|ERROR=${JSON.stringify(error)}`);
-    throw new Error(
-      `Error while fetching the notification: ${JSON.stringify(error)}`
-    );
-  }
-
-  const maybeEmailNotification = errorOrMaybeNotification.right;
-
-  if (O.isNone(maybeEmailNotification)) {
-    // it may happen that the object is not yet visible to this function due to latency
-    // as the notification object is retrieved from database and we may be hitting a
-    // replica that is not yet in sync - throwing an error will trigger a retry
-    context.log.warn(`${logPrefix}|RESULT=NOTIFICATION_NOT_FOUND`);
-    throw new Error(`Notification not found`);
-  }
-
-  const errorOrEmailNotification = EmailNotification.decode(
-    maybeEmailNotification.value
-  );
-
-  if (E.isLeft(errorOrEmailNotification)) {
-    // The notification object is not compatible with this code
-    const error = readableReport(errorOrEmailNotification.left);
-    context.log.error(`${logPrefix}|ERROR`);
-    context.log.verbose(`${logPrefix}|ERROR_DETAILS=${error}`);
-    return EmailNotificationResult.encode({
-      kind: "FAILURE",
-      reason: "DECODE_ERROR"
-    });
-  }
-
-  const emailNotification = errorOrEmailNotification.right.channels.EMAIL;
-
-  const documentHtml = await generateDocumentHtml(
-    content.subject,
-    content.markdown,
-    senderMetadata
+) =>
+  withJsonInput(
```

Comment:
This line is the actual change: handler function has been wrapped by withJsonInput decorator